symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -164,60 +174,70 @@ public TemplateEngine templates() {
-   * @param stream     the stream to send the message to
-   * @param template   the template name to be used to produce the message
-   * @param parameters the parameters to pass to the template to produce the message to be sent
+   * @param streamId the ID of the stream to send the message to
+   * @param message  the message payload in MessageML
    * @return a {@link V4Message} object containing the details of the sent message
    * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
+   * @deprecated this method will be replaced by {@link MessageService#send(String, Message)}
    */
-  public V4Message send(@Nonnull V4Stream stream, @Nonnull String template, Object parameters)
-      throws TemplateException {
-    return send(stream.getStreamId(), templateResolver.resolve(template).process(parameters));
-  }
-
-  /**
-   * Sends a templated to the stream ID passed in parameter.
-   *
-   * @param streamId   the ID of the stream to send the message to
-   * @param template   the template name to be used to produce the message
-   * @param parameters the parameters to pass to the template to produce the message to be sent
-   * @return a {@link V4Message} object containing the details of the sent message
-   * @throws TemplateException
-   */
-  public V4Message send(@Nonnull String streamId, @Nonnull String template, Object parameters)
-      throws TemplateException {
-    return send(streamId, templateResolver.resolve(template).process(parameters));
+  @Deprecated
+  public V4Message send(@Nonnull String streamId, @Nonnull String message) {
+    return executeAndRetry("send", () -> messagesApi.v4StreamSidMessageCreatePost(
+        streamId,
+        authSession.getSessionToken(),
+        authSession.getKeyManagerToken(),
+        message,
+        null,
+        null,
+        null,
+        null
+    ));
   }
 
   /**
    * Sends a message to the stream ID passed in parameter.
    *
-   * @param streamId the ID of the stream to send the message to
-   * @param message  the message payload in MessageML
+   * @param streamId    the ID of the stream to send the message to
+   * @param message     the message to send to the stream
    * @return a {@link V4Message} object containing the details of the sent message
    * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
    */
-  public V4Message send(@Nonnull String streamId, @Nonnull String message) {
+  public V4Message send(@Nonnull String streamId, @Nonnull Message message) {
+    File tempFile = this.createTemporaryAttachmentFile(message.getAttachment());
     return executeAndRetry("send", () -> messagesApi.v4StreamSidMessageCreatePost(
         streamId,
         authSession.getSessionToken(),
         authSession.getKeyManagerToken(),
-        message,
-        null,
-        null,
+        message.getContent(),
+        message.getData(),
         null,
+        tempFile,
```

Comment:
`this.createTemporaryAttachmentFile(message.getAttachment())` directly here

Reply:
I don't know why yet but if I use `this.createTemporaryAttachmentFile(message.getAttachment())`, method will throw a `RuntimeException` instead of `MessageCreationException` and the unittest will fail

Reply:
because the retry throws every `Throwable` as `RuntimeException`

Reply:
And I would create the temp file outside the retry to avoid new file created if the retry recovery is triggered